// Copyright 2016 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "vector3.proto";
import "vector4.proto";

message PropertyBuffer {
	enum Type { VEC3 = 1; VEC4 = 2; }
	string name = 1;
	string display_name = 2;
	Type type = 3;
	optional Vector3Buffer vec3 = 4;
	optional Vector4Buffer vec4 = 5;
}

message ShaderProgramBuffer {
	repeated PropertyBuffer properties = 1;
}
